source/Halibut/ServiceModel/PendingRequestQueue.cs

```diff
@@ -68,6 +75,17 @@ public bool IsEmpty
             }
         }
 
+        public int Count
+        {
+            get
+            {
+                lock (sync)
```

Comment:
Why was this required?

Reply:
oh similar to is empty.

Reply:
Yeah, I needed to know in the tests when the queue had filled up completely before we could start dequeueing.